core/frontend/src/tile/RealityTile.ts

```diff
@@ -231,60 +245,186 @@ export class RealityTile extends Tile {
 
       this.selectRealityChildren(context, args, traversalDetails);
-      if (this.isReady && (traversalDetails.childrenLoading || 0 !== traversalDetails.queuedChildren.length)) {
-        const minimumVisibleFactor = .25;     // If the tile has not yet been displayed in this viewport -- display only if it is within 25% of visible. Avoid overly tiles popping into view unexpectedly (terrain)
 
-        if (visibility > minimumVisibleFactor || this._everDisplayed)
+      // Children are not ready: use this tile to avoid leaving a hole
+      traversalDetails.shouldSelectParent = traversalDetails.shouldSelectParent || traversalDetails.queuedChildren.length !== 0;
+
+      if (traversalDetails.shouldSelectParent) {
+        // If the tile has not yet been displayed in this viewport -- display only if it is visible enough. Avoid overly tiles popping into view unexpectedly (terrain)
+        if (visibility > this.minimumVisibleFactor() || this._everDisplayed) {
           context.selectOrQueue(this, args, traversalDetails);
+        }
       }
     }
   }
 
-  public purgeContents(olderThan: BeTimePoint): void {
-    // Discard contents of tiles that have not been "used" recently, where "used" may mean: selected/preloaded for display or content requested.
+  // Attempt to select the children of a tile in case they could be displayed while this tile is loading. This does not take into account visibility.
+  protected selectRealityChildrenAsFallback(context: TraversalSelectionContext, args: TileDrawArgs, traversalDetails: TraversalDetails){
+    const childrenReady = this.getLoadedRealityChildren(args);
+
+    if (childrenReady) {
+      context.select(scratchLoadedChildren, args);
+      traversalDetails.shouldSelectParent = false;
+    }
+
+    scratchLoadedChildren.length = 0;
+  }
+
+  // Recurse through children to select them normally
+  protected selectRealityChildren(context: TraversalSelectionContext, args: TileDrawArgs, traversalDetails: TraversalDetails) {
+
+    // Load children if not yet requested
+    const childrenLoadStatus = this.loadChildren(); // NB: asynchronous
+
+    // Children are not ready yet
+    if (childrenLoadStatus === TileTreeLoadStatus.Loading) {
+      args.markChildrenLoading();
+      traversalDetails.shouldSelectParent = true;
+      return;
+    }
+
+    if (this.realityChildren !== undefined) {
+      // Attempt to select the children
+      const traversalChildren = this.realityRoot.getTraversalChildren(this.depth);
+      traversalChildren.initialize();
+
+      for (let i = 0; i < this.children!.length; i++)
+        this.realityChildren[i].selectRealityTiles(context, args, traversalChildren.getChildDetail(i));
+
+      traversalChildren.combine(traversalDetails);
+    }
+  }
+
+  public purgeContents(olderThan: BeTimePoint, useProtectedTiles: boolean): void{
+    const tilesToPurge = new Set<RealityTile>();
+
+    // Get the list of tiles to purge
+    if (useProtectedTiles)
```

Comment:
I split it in two function because I thought it would make it more readable. The version without protected tiles could easily be merged with the version with protected tiles, but it would make the behavior less obvious.

Should we keep it like this or merge them?

Reply:
I vote for keeping behavior "more obvious".